hbw/app/assets/javascripts/hbw/translator.js.coffee.erb

```diff
@@ -1,13 +1,25 @@
 modulejs.define 'HBWTranslator', ['jQuery'], (jQuery) ->
   translations = <%= I18n.t('hbw.js').to_json %>
 
-  translate: (key) ->
+  translate: (key, vars = {}) ->
     current = translations
 
     for part in key.split('.')
       if jQuery.isPlainObject(current) and part of current
-        current = current[part]
+        current = @interpolateString(current[part], vars)
       else
         throw new Error('Cannot find ' + key + ' in translations')
 
     current
+
+  interpolateString: (str, vars) ->
+    if typeof str == 'string'
+      str.replace(/%{([^{}]*)}/g, (full_match, var_name) ->
+        subst = vars[var_name]
+        if typeof subst == 'string' || typeof subst == 'number'
+          subst
+        else
+          throw new Error('Undefined variable ' + var_name)
```

Comment:
Putting exceptions as ```else``` is a not a great practice.

Better put a check before happy path. Something like
```
if smth_bad
  throw new Error

subst
```